nilearn/surface/surface.py

```diff
@@ -895,6 +897,65 @@ def load_surf_mesh(surf_mesh):
     return mesh
 
 
+def load_surface(surface):
+    """Loads a surface.
+
+    Parameters
+    ----------
+    surface : List of numpy.ndarray or Surface
+        Surface to be loaded. This can be passed as:
+
+        - a list of two files (valid formats are .gii
+        .gii.gz or Freesurfer specific files such as
+        .orig, .pial, .sphere, .white, .inflated) containing:
+            - a surface mesh geometry
+            - surface data
+        - three Numpy arrays organized in a list with coordinates,
+        faces, and data in this specific order
+        - a length 2 tuple or a namedtuple with the fields "mesh" and "data"
+        - a length 3 tuple or a namedtuple with the fileds "coordinates",
+        "faces", and "data"
+        - a Surface object with "mesh" and "data" attributes.
+
```

Comment:
To avoid having to list all combinations of types for mesh and data I would
suggest something like:
```
surface can be:
  - a nilearn.surface.Surface
  - a sequence (mesh, data) where:
    - mesh can be:
      - a nilearn.surface.Mesh
      - a path to .gii or .gii.gz etc.
      - a sequence of two numpy arrays, the first containing vertex coordinates
        and the second containing triangles.
    - data can be:
      - a path to .gii or .gii.gz etc.
      - a numpy array with shape (n_vertices,) or (n_time_points, n_vertices)
```

also is it possible to pass a single gifti file containing the mesh and the data?

Reply:
Thanks for the reviews @jeromedockes ! :+1: 
I agree, this is much cleaner and I'm not sure that having the possibility to load a surface with a flat sequence of length 3 would be super useful indeed. I incorporated your suggestions and updated the tests.

>also is it possible to pass a single gifti file containing the mesh and the data?

I don't think you can do that, but I don't know much about the gifti format so I might be wrong here...